### ISSUE TITLE
[Feature Request] MariaDB

### ISSUE THREAD

**Issue by Kalzem** [TablePlus/DBngin#48]
Hello,

It would be great if MariaDB could be added to the list of engines :)

Thank you

**Comment by tholu** [comment]
Would use it and even buy TablePlus if MariaDB would be available. Unfortunately, MySQL8 is not fully compatible to MariaDB.

**Comment by BuddhiAbeyratneAL** [comment]
How tough would it be to add this addition @huyphams if someone wanted to try and do it as a contribution ?

**Comment by eleftrik** [comment]
Any plan to add this feature? Or DBngin development has ended? Thanks you!

**Comment by mludi** [comment]
Is this project still alive or should I move to another solution?

**Comment by encodiaweb** [comment]
> Is this project still alive or should I move to another solution?

I'm asking myself the same question... And up now I've not found any similar solution

**Comment by BuddhiAbeyratneAL** [comment]
Don't think development has ended but think that its not high priority. Probs why the request is not being responded to.

**Comment by encodiaweb** [comment]
> Don't think development has ended but think that its not high priority. Probs why the request is not being responded to.

I agree with stating that nothing is due, time is not enough, etc. But after more than 2 years I think that replying to this issue with "won't fix" or "we'll add this in N months" would be nice.

**Comment by mansurmetocrm** [comment]
bump still need mariadb what's the problem to add as a seperate driver ?

**Comment by kaiserkiwi** [comment]
Started using DBngin and would love to drop Homebrew. Any chance this will be worked on? Or is there a maintained alternative to this?

**Comment by thosky** [comment]
I cannot use DBngin because it's not supporting MariaDB. Why is this still a problem?

**Comment by sebfie** [comment]
We also need it here!

**Comment by AdyGCode** [comment]
Upvote to this.

Many of us are moving to - or have moved to - MariaDB so we can keep away from influence Oracle ;-)

**Comment by musa11971** [comment]
On Nov 24, 2023 TablePlus announced the following on Twitter:

![Screenshot 2024-01-16 at 22 55 22](https://github.com/TablePlus/DBngin/assets/21341801/7bf735ee-3784-476a-8d39-2090fc24d8c1)
https://twitter.com/TablePlus/status/1728119754647388564

---

No further update for now it seems... let's hope it gets added soon! 🎉

**Comment by besingamkb** [comment]
4 years and still not added? I think they have some partnership from oracle or something that preventing them to add mariadb here..

**Comment by kaiserkiwi** [comment]
This app had no commits for two years now. I guess it's just abandoned. No need for conspiracy theories here. 

It would probably wise to fork it and add it yourself if you still need it. 😕

**Comment by kaiserkiwi** [comment]
No need for conspiracy theories here. The mentioned post is not that old. This tool is free and as far as I know only one person maintains it.

It would probably wise to fork it and add it yourself if you're in hurry.

**Comment by huyphams** [comment]
This is the issue tracking, it has nothing todo the codebase. The most recently commit was a week ago

**Comment by huyphams** [comment]
We need recompile the whole mariadb project in order to add it to dbngin, it is lunar new year so our team is on holiday.
@kaiserkiwi we have 2 people are working on dbngin part time.

**Comment by huyphams** [comment]
It is free product. We do not receive money from any org. 100% fund goes from https://tableplus.com (hosting, CDN, and human resource...)

**Comment by mansurmetocrm** [comment]
> No need for conspiracy theories here. The mentioned post is not that old. This tool is free and as far as I know only one person maintains it.
> 
> It would probably wise to fork it and add it yourself if you're in hurry.

not open source doe, would happy to fork and try to add.

**Comment by WeAreHaus** [comment]
@huyphams Any news on a DBngin version supporting MariaDB. We would love to use your product!

**Comment by jonjieviduya** [comment]
Any updates for MariaDb support?

**Comment by kevinpickett** [comment]
Any updates on this? Everything else announced was released but it looks like MariaDB support was dropped. Is there some technical reason MariaDB is more difficult to integrate? Every DB app that has announced local MariaDB support has failed to follow through on that promise from what I have seen. It is difficult not to get conspiratorial when this pattern continues to play out.

**Comment by LeonLiutas** [comment]
Any updates on support for MariaDB? We would love to adopt Herd in our development team.

**Comment by Stetzon** [comment]
![Image](https://github.com/user-attachments/assets/f74254e6-c3ab-4ef8-b09f-bfaeeaae1da7)

**Comment by abdorizak** [comment]
any update?

**Comment by musa11971** [comment]
Hi all, I’ve been following this thread and noticed several people are looking for MariaDB support. I’m the developer of [SnapDB](https://snapdb.app/), a macOS database management tool that fully supports MariaDB (including dependency-free installation, management, and configuration). If anyone here is still searching for a workaround, feel free to check it out—it might save you some time. Happy to answer questions!

<img width="1012" alt="Image" src="https://github.com/user-attachments/assets/2fbf5d9b-d394-42c8-8257-e404604fbc7e" />

**Comment by huyphams** [comment]
Will be ready in the next release for free

<img width="618" alt="Image" src="https://github.com/user-attachments/assets/1ed437da-17c2-4e26-ad44-ea7134825862" />

**Comment by huyphams** [comment]
If you have a chance, you can download pre-release here which supports MariaDB 
https://www.dropbox.com/scl/fi/b1qqg50wtjuhzzssvku2h/DBngin.zip?rlkey=wguwz3tu0ydriimppod1m3nq0&dl=0

**Comment by AdyGCode** [comment]
Download ok

On execution, there is no options in the MariaDB  version download.

I do have MariaDB installed via Homebrew so not sure if that will be part of the issue.

Ady




Ady Gould
Educator, Coder, Hockey Goalie, AFOL, Geek & Nerd

**Comment by huyphams** [comment]
Are you using macOS Intel @AdyGCode ?

<img width="562" alt="Image" src="https://github.com/user-attachments/assets/cfccb74e-15c9-4f6b-968a-e8f0edcbc795" />

**Comment by AdyGCode** [comment]
Sorry Henry...
I should have thought about telling you more..
End of work day so just call it a brain fart...

Mac Pro - Intel - 2019 - 32GB
Sequoia 15.2



Ady Gould
Educator, Coder, Hockey Goalie, AFOL, Geek & Nerd

**Comment by huyphams** [comment]
Thanks, I'm compiling it for Intel, and will be ready by tomorrow, and you don't have to redownload the build, just wait for it

**Comment by huyphams** [comment]
DBngin 74 which supports MariaDB is released!

**Comment by mludi** [comment]
Thank you! 😍

**Comment by araujo0205** [comment]
![Image](https://github.com/user-attachments/assets/a0983595-067c-4c7c-971e-f6680689b348)
I updated but have no option to select mariadb version. 
I try reinstall but no solve error.

**Comment by spectabile** [comment]
> I updated but have no option to select mariadb version. I try reinstall but no solve error.

Same here.
MySQL 8, PostgreSQL, Redis are OK, the select shows the versions.

MacOS Monterey 12.7.6, Intel Core i7
